packages/webapp/pages/plus/index.tsx

```diff
@@ -31,6 +37,12 @@ const PlusPage = (): ReactElement => {
     [productPrices?.data.currencyCode, productPrices?.data?.details?.lineItems],
   );
 
+  useEffect(() => {
+    if (!productPrices) {
+      refetchPlanTypes();
+    }
+  }, [productPrices, refetchPlanTypes]);
+
```

Comment:
Why do we need to refetch?
It will still fail if you not enrolled in experiment right?

I assumed we would just not check for feature flag on that query maybe?
If we go this way we also need to modify the selection state in the ref.

Reply:
It's how we can force fetch it if the query is disabled

Reply:
Also maybe better way of refetching would be to add user to query key so it automatically refetches when it hydrates?

Reply:
With `!isPlus` it would still fetch pricingIds if you're not in the experiment, a wasted request IMO

Reply:
Ideally we should just fetch the prices once the plus page is visited, no?

Reply:
Yeah, it triggered fetch on any page you visited

Reply:
Maybe it wasn't intentional?

Reply:
It triggers a fetch because it's a global context. Same with `initializePaddle` actually, it is ran on any page you visit

Reply:
![image](https://github.com/user-attachments/assets/f8c1b1fa-cd9a-4621-b8e7-a6271285864c)
![image](https://github.com/user-attachments/assets/89be32ea-57ff-4fe1-9a1b-d7ff650f5ecc)

Reply:
Yeah, probably. Maybe we should move PaymentContext into `getPlusLayout` instead of having it globally?

Reply:
Hmm we can, initially I thought we could use it for other stuff, but let's do that rather.
Can you do it?

Reply:
Fixing it